examples/container-hook/main.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	ocispec "github.com/opencontainers/runtime-spec/specs-go"
+	api "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/container-hook"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+)
+
+var (
+	outputList = flag.String("output", "add,remove", "comma-separated list of events to print [add,remove,config]")
+
+	outputAdd    bool
+	outputRemove bool
+	outputConfig bool
+
+	hookPreStart = flag.String("prestart", "", "command to run in the PreStart hook")
+	hookPostStop = flag.String("poststop", "", "command to run in the PostStop hook")
+	env          = flag.String("env", "", "the environ")
+	dir          = flag.String("dir", "", "dir")
+	timeout      = flag.String("timeout", "10s", "timeout")
+
+	timeoutDuration time.Duration
+
+	publishKubernetesEvent = flag.Bool("publish-kubernetes-event", false, "Publish an event using the Kubernetes Event API")
+	kubeconfig             = flag.String("kubeconfig", "", "kubeconfig")
+	node                   = flag.String("node", "", "Node name")
+
+	client *kubernetes.Clientset
+)
```

Comment:
I know the existing container has this, but it's not clear to me why using kubernetes is beneficial for this example. IMO it creates confusion as people could think kubernetes is required for this package, and it's not. I'd personally would remove the k8s part to make it clear it's about containers, but it's totally up to you.